cdap-data-fabric/src/main/java/co/cask/cdap/data2/audit/AuditPublishers.java

```diff
@@ -51,6 +58,13 @@ public static void publishAccess(@Nullable AuditPublisher publisher, EntityIdCom
       return;
     }
 
+    AccessedEntityInfo accessedEntityInfo = new AccessedEntityInfo(entityId, accessType);
+    EntityIdCompatible accessorId = CACHE_AUDIT_LOGS.getIfPresent(accessedEntityInfo);
+    if (accessorId != null && Objects.equals(accessorId, accessor)) {
```

Comment:
If you pass `accessorId` in the key, and return `boolean` from cache load method then we'll take care of both the above points, right?